bluesky/callbacks/stream.py

```diff
@@ -0,0 +1,155 @@
+import logging
+import time as ttime
+from collections import Iterable, ChainMap
+
+import jsonschema
+import numpy as np
+from event_model import DocumentNames, schemas
+
+from .core import CallbackBase
+from ..run_engine import Dispatcher
+from ..utils import new_uid
+
+logger = logging.getLogger(__name__)
+
+
+class LiveDispatcher(CallbackBase):
+    """Create a secondary event stream of processed data"""
+    def __init__(self):
+        # Public dispatcher for callbacks
+        self.dispatcher = Dispatcher()
+        # Local caches for internal use
+        self.seq_count = 0  # Maintain our own sequence count for this stream
+        self.raw_descriptors = dict()  # Store raw descriptors for use later
+        self._stream_start_uid = None  # Generated start doc uid
+        self._descriptors = dict()  # Dictionary of sent descriptors
+
+    def start(self, doc, _md=None):
+        """Receive a start document, re-emit it for this event stream"""
+        self._stream_start_uid = new_uid()
+        _md = _md or dict()
+        # Create a new start document with a new uid, start time, and the uid
+        # of the original start document. Preserve the rest of the metadata
+        # that we retrieved from the start document
+        md = ChainMap({'uid': self._stream_start_uid,
+                       'original_run_uid': doc['uid'],
+                       'time': ttime.time()},
+                      _md, doc)
+        # Dispatch the start document for anyone subscribed to our Dispatcher
+        self.emit(DocumentNames.start, dict(md))
+        super().start(doc)
+
+    def descriptor(self, doc):
+        """Store a descriptor"""
+        self.raw_descriptors[doc['uid']] = doc
+        super().descriptor(doc)
+
+    def process_event(self, doc, stream_name='primary',
+                      id_args=None, config=None):
+        """
+        Receive an event document
```

Comment:
Please include what the event is revived from.   Keep the to / from receive/send verbiage clear seems to the hardest documentation challenge for any networky type project :wink:

Reply:
Are you referring to keeping the `uid` from the original `Event` document or the entire thing? I thought about making this mandatory in the call, but I didn't want to enforce that a single processed event from the pipeline needed to be linked with a single raw event.

Reply:
I think that "include what the event is revived from" is supposed to say "include what the event is _received_ from" in the docstring -- something like:

```
Receive an event document from the emit method.
```